knowhere/feder/HNSW.h

```diff
@@ -0,0 +1,203 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#pragma once
+
+#include <memory>
+#include <tuple>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+#include "knowhere/feder/IDSet.h"
+
+namespace knowhere::feder::hnsw {
+
+///////////////////////////////////////////////////////////////////////////////
+// index view
+struct NodeInfo {
+    int64_t id_;
+    std::vector<int64_t> links_;
+
+    NodeInfo() = default;
+
+    NodeInfo(const int64_t id, const std::vector<int64_t>& links) = delete;
```

Comment:
Can I ask why we need to delete this here? If we simply want to just use `NodeInfo()` as constructor, we can add `explicit` before it, so the implicit convert like `NodeInfo{a, b}` will be forbidden.